queries.sql

```diff
@@ -19,3 +19,47 @@ SELECT * FROM animals WHERE name != 'Gabumon'
 
 SELECT * FROM animals
 WHERE weight_kg >= 10.4 and weight_kg <= 17.3
+
+BEGIN;
+UPDATE animals SET species = 'unspecified'
+SELECT * FROM animals
+ROLLBACK
+SELECT * FROM animals
+
+-- Updating species 
+UPDATE animals SET species = 'digimon 'WHERE name LIKE '%mon'
+UPDATE animals SET species = 'pokemon' WHERE name NOT LIKE '%mon'
+SELECT * FROM animals
+COMMIT
+
+-- Deleting records
+BEGIN
+DELETE FROM animals
+ROLLBACK
+SELECT * FROM animals
+BEGIN
+DELETE FROM animals WHERE date_of_birth > '2022-01-01'
+SAVEPOINT savepoint1
+UPDATE animals SET weight_kg = weight_kg * -1
+ROLLBACK TO SAVEPOINT savepoint1
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg <0
+COMMIT
+
+--      Queries
+
+SELECT COUNT(*) FROM animals
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0
+SELECT AVG(weight_kg) FROM animals
+
+SELECT name, id FROM animals
+ORDER BY escape_attempts DESC
+LIMIT 1
```

Comment:
- [ ] Please add the correct query to answer the question 👍
* `Who escapes the most, neutered or not neutered animals?`